### PR TITLE
Fix documentation around using xtrabackup as backup engine

### LINKED ISSUE #261: xtrabackup is not documented
See: https://vitess.io/docs/user-guides/backup-and-restore/ -- there is no mention of the xtrabackup plugin.

### PR DESCRIPTION
**Pull request by hwang381**
Fix documentation around using xtrabackup as backup engine

Resolves #261

### CODE REVIEW

File: content/en/docs/user-guides/backup-and-restore.md [hunk 1]
```diff
@@ -136,13 +136,20 @@ do this for all the instances it creates by adding `--scopes storage-rw` to the
...
+Current, we have plugins for:
+
+* Builtin: Copy all the database files into specified storage. This is the default.
+* Percona Xtrabackup
+
 Run the following vtctl command to create a backup:
 
 ``` sh
-vtctl Backup <tablet-alias>
+vtctl Backup <tablet-alias> -backup_engine_implementation=builtin
```
Comment (deepthi) [inline content/en/docs/user-guides/backup-and-restore.md]: There is no such option to the `Backup` command.

File: content/en/docs/user-guides/backup-and-restore.md [hunk 2]
```diff
@@ -21,7 +21,6 @@ Currently, we have plugins for:
 * Google Cloud Storage
 * Amazon S3
 * Ceph
-* Xtrabackup
```
Comment (deepthi) [inline content/en/docs/user-guides/backup-and-restore.md:24]: +1

File: content/en/docs/user-guides/backup-and-restore.md [hunk 3]
```diff
@@ -49,17 +48,6 @@ access to the location where you are storing backups.
         </ul>
       </td>
     </tr>
-    <tr>
```
Comment (deepthi) [inline content/en/docs/user-guides/backup-and-restore.md]: This block should be added back.

File: content/en/docs/user-guides/backup-and-restore.md [hunk 4]
```diff
@@ -136,13 +136,20 @@ do this for all the instances it creates by adding `--scopes storage-rw` to the
 
 ## Creating a backup
 
+Vitess supports multiple ways to generate data backups. This is called a Backup engine, which is a [pluggable interface](https://github.com/vitessio/vitess/blob/master/go/vt/mysqlctl/backupengine.go)
+
+Current, we have plugins for:
```
Comment (deepthi) [inline content/en/docs/user-guides/backup-and-restore.md]: Current -> Currently

File: content/en/docs/user-guides/backup-and-restore.md [hunk 5]
```diff
@@ -136,13 +136,20 @@ do this for all the instances it creates by adding `--scopes storage-rw` to the
...
+
 Run the following vtctl command to create a backup:
 
 ``` sh
-vtctl Backup <tablet-alias>
+vtctl Backup <tablet-alias> -backup_engine_implementation=builtin
 ```
 
-If the engine is `builtin`, in response to this command, the designated tablet performs the following
+If `backup_engine_implementation` is `builtin`, in response to this command, the designated tablet performs the following
```
Comment (deepthi) [inline content/en/docs/user-guides/backup-and-restore.md]: Revert this change.

File: content/en/docs/user-guides/backup-and-restore.md [hunk 6]
```diff
@@ -167,7 +174,7 @@ sequence of actions:
    be behind on replication, and not used by vtgate for serving until it catches
    up.
 
-If the engine is `xtrabackup`, we do not do any of the above. The tablet can
+If `backup_engine_implementation` is `xtrabackup`, we do not do any of the above. The tablet can
```
Comment (deepthi) [inline content/en/docs/user-guides/backup-and-restore.md]: Revert this change.

**Review by deepthi — CHANGES_REQUESTED** [review]
Thank you for creating this PR. We definitely appreciate the help.

**Comment by hwang381** [comment]
All comments have been addressed @deepthi

**Comment by deepthi** [comment]
@hwang381 you have to fix the DCO check. You can do that by rebasing and fixing each commit comment to have a signoff line, and then force-push.

File: content/en/docs/user-guides/backup-and-restore.md [hunk 7]
```diff
@@ -120,6 +126,18 @@ access to the location where you are storing backups.
...
+      <td>For the <code>xtrabackup</code> backup engine, required user that xtrabackup will use to connect to the database server. This user must have all necessary privileges. For details, please refer to xtrabackup documentation.</td>
+    </tr>
+    <tr>
+      <td><code>xtrabackup_backup_flags</code></td>
+      <td>For the <code>xtrabackup</code> backup engine, flags to pass to backup command. These should be space separated and will be added to the end of the command.</td>
+    </tr>
+     <tr>
+      <td><code>xtrabackup_stream_mode</code></td>
+      <td>For the <code>xtrabackup</code> backup engine, which mode to use if streaming, valid values are <code>tar</code> and <code>xbstream</code>. Defaults to <code>tar</code>.</td>
+    </tr>
```
Comment (deepthi) [inline content/en/docs/user-guides/backup-and-restore.md:156]: There are 3 other flags:
xtrabackup_root_path
xtrabackup_stripes
xtrabackup_stripe_block_size.

Would you be able to add these to the docs?

**Review by deepthi — APPROVED** [review]
LGTM

**Pull request merged**